scripts/format.py

```diff
@@ -1,467 +1,299 @@
-    for fname in changed_files:
-        print(fname)
-else:
-    print(action + " all files")
-
-if confirm and not check_only:
-    print("The files listed above will be reformatted.")
-    result = input("Continue with changes (y/n)?\n")
-    if result != 'y':
-        print("Aborting.")
-        exit(0)
-
-format_commands = {
-    '.cpp': cpp_format_command,
-    '.ipp': cpp_format_command,
-    '.c': cpp_format_command,
-    '.hpp': cpp_format_command,
-    '.h': cpp_format_command,
-    '.hh': cpp_format_command,
-    '.cc': cpp_format_command,
-    '.txt': cmake_format_command,
-    '.py': 'black --quiet - --skip-string-normalization --line-length 120 --stdin-filename',
-    '.java': cpp_format_command,
-}
-
-difference_files = []
-
-header_top = "//===----------------------------------------------------------------------===//\n"
-header_top += "//                         DuckDB\n" + "//\n"
-header_bottom = "//\n" + "//\n"
-header_bottom += "//===----------------------------------------------------------------------===//\n\n"
-base_dir = os.path.join(os.getcwd(), 'src/include')
+def format_directory(directory):
+    files = []
+    for path in Path(directory).rglob("*"):
+        if path.is_file() and can_format_file(str(path)):
+            files.append(str(path))
+    return files
 
 
-def get_formatted_text(f, full_path, directory, ext):
-    if not can_format_file(full_path):
+# Formatting logic
+def format_file(full_path, check_only, force, silent):
+    ext = Path(full_path).suffix
+    if ext not in FORMAT_COMMANDS:
         if not force:
-            print(
-                "File "
-                + full_path
-                + " is not normally formatted - but attempted to format anyway. Use --force if formatting is desirable"
-            )
-            exit(1)
-    if f == 'list.hpp':
-        # fill in list file
-        file_list = [
-            os.path.join(dp, f)
-            for dp, dn, filenames in os.walk(directory)
-            for f in filenames
-            if os.path.splitext(f)[1] == '.hpp' and not f.endswith("list.hpp")
-        ]
-        file_list = [x.replace('src/include/', '') for x in file_list]
-        file_list.sort()
-        result = ""
-        for x in file_list:
-            result += '#include "%s"\n' % (x)
-        return result
-
-    if ext == ".hpp" and directory.startswith("src/include"):
-        with open_utf8(full_path, 'r') as f:
-            lines = f.readlines()
-
-        # format header in files
-        header_middle = "// " + os.path.relpath(full_path, base_dir) + "\n"
-        text = header_top + header_middle + header_bottom
-        is_old_header = True
-        for line in lines:
-            if not (line.startswith("//") or line.startswith("\n")) and is_old_header:
-                is_old_header = False
-            if not is_old_header:
-                text += line
-
-    if ext == '.test' or ext == '.test_slow' or ext == '.test_coverage' or ext == '.benchmark':
```

Comment:
If you're touching this file, you might want to preserve the existing behavior
With removing all of these lines I doubt that this behaves the same.

Reply:
Does cmake-format actually have the ability to format `.test` files?

```
cmake-format -i test/planner/projection_binding.test
...
ValueError: Unexpected WORD token at 5:10, expecting l-paren, got 'ok'
```

Reply:
But your overall observation is right - substantial parts of the file have been rewritten. Only the argparse/top level control flow can be described as a refactor.

Reply:
I missed that. What if we move the formatting logic here to a tool of its own and invoke it similar to other tools?

Reply:
Updated implementation tries to stick to the original algorithm as closely as possible. A small number of deltas were found and included in a separate commit for your review.